query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterIndexingServiceProvider.java

```diff
@@ -245,6 +246,15 @@ public void onFilterResult(Object userContext, Object instance, Object eventType
                event = clone;
             }
             for (DelegatingCacheEntryListenerInvocation<K, V> invocation : invocations) {
+               if (event.isPre()) {
```

Comment:
I thought about this more.  The listener always has an observation.  Maybe we can just do?
```
if (invocation.getObservation().shouldInvoke(event.isPre())) {
   invocation.invokeNoChecks(event, false, filterAndConvert);
}
```

We may have to change https://github.com/infinispan/infinispan/pull/3643/files#diff-896a626d3ff4076e967def69ba12d01dR329 to return BOTH as well maybe.